components/sup/src/manager/service_updater/package_update_worker.rs

```diff
@@ -103,8 +110,11 @@ impl PackageUpdateWorker {
         }
     }
 
-    /// Use the service spec's package ident to search for packages.
-    pub async fn update(&self) -> PackageIdent { self.update_to(self.ident.clone()).await }
+    /// Use the service spec's package ident to search for packages taking into account the update
+    /// condition.
+    pub async fn update(&self, condition: UpdateCondition) -> PackageIdent {
```

Comment:
Should the `condition` be part of the update worker's internal state? It and the ident both come from the service spec, for instance. It's also part of the `RollingUpdateWorker`.

(Same goes for the `update_to` method.)

Reply:
This is much better, thanks! Addressed in [e31171](https://github.com/habitat-sh/habitat/pull/7513/commits/e31171b0761c41c07923152087f0ca83409dcbc0)